test/functional/container.spec.ts

```diff
@@ -5,166 +5,134 @@ import { ContainerRequest } from "../../dist-esm/client/Container/ContainerReque
+      assert(results.length > 0, "number of results for the query should be > 0");
+
+      const { resources: ranges } = await container.readPartitionKeyRanges().fetchAll();
+      assert(ranges.length > 0, "container should have at least 1 partition");
+
+      // Replacing indexing policy is allowed.
+      containerDef.indexingPolicy.indexingMode = IndexingMode.lazy;
+      const { resource: replacedContainer } = await container.replace(containerDef);
+      assert.equal("lazy", replacedContainer.indexingPolicy.indexingMode);
 
-    it("nativeApi Should do container CRUD operations successfully name based", async function() {
+      // Replacing partition key is not allowed.
       try {
-        await containerCRUDTest(false);
+        containerDef.partitionKey = { paths: ["/key"], kind: PartitionKind.Hash };
+        await container.replace(containerDef);
+        assert.fail("Replacing paritionkey must throw");
       } catch (err) {
-        throw err;
+        const badRequestErrorCode = 400;
+        assert.equal(err.code, badRequestErrorCode, "response should return error code " + badRequestErrorCode);
+      } finally {
+        containerDef.partitionKey = containerDefinition.partitionKey; // Resume partition key
       }
-    });
-
-    it("nativeApi Should do elastic container CRUD operations successfully name based", async function() {
+      // Replacing id is not allowed.
       try {
-        await containerCRUDTest(true);
+        containerDef.id = "try_to_replace_id";
+        await container.replace(containerDef);
+        assert.fail("Replacing container id must throw");
       } catch (err) {
-        throw err;
+        const notFoundErrorCode = 400;
+        assert.equal(err.code, notFoundErrorCode, "response should return error code 404");
       }
-    });
 
-    it("nativeApi container with bad partition key definition name based", async function() {
+      // read container
+      containerDef.id = containerDefinition.id; // Resume Id.
+      const { resource: readcontainer } = await container.read();
+      assert.equal(containerDefinition.id, readcontainer.id);
+
+      // delete container
+      await container.delete();
+
+      // read container after deletion
       try {
-        await badPartitionKeyDefinitionTest(true);
+        await container.read();
+        assert.fail("Must fail to read container after delete");
       } catch (err) {
-        throw err;
+        const notFoundErrorCode = 404;
+        assert.equal(err.code, notFoundErrorCode, "response should return error code 404");
       }
+    };
+
+    it("Default partition key", async function() {
+      await containerCRUDTest();
+    });
+
+    it("Custom partition key", async function() {
+      await containerCRUDTest("/id");
     });
 
-    it("nativeApi container with bad partition key definition name based", async function() {
+    it("Bad partition key definition", async function() {
```

Comment:
Not sure if we need a test to check that hash isn't allowed anymore. This kind of testing is prone to breaking.